### ISSUE TITLE
Questions about reward model training and evaluation details

### ISSUE THREAD

**Issue by CiaranZhou** [facebookresearch/sweet_rl#4]
Hi Yifei,

Thanks for the great work! This is a very interesting paper.

I'm currently trying to reproduce the results in the paper, but I ran into a few issues I’d appreciate your clarification on:

1. Regarding the training of the step-level reward model, the README mentions the model path as `/PATH/TO/8BLLAMA`. Just to confirm, is this referring to `Llama-3.1-8B` or `Llama-3.1-8B-Instruct`?

2. In the data sampling step, I noticed the `data_fraction` parameter is set to 0.1. Could you elaborate on the motivation for using this value?

3. For the `backend` task, the paper states that the test set contains 1k examples, but the corresponding dataset on Huggingface seems to include 10k. Could you clarify what the actual test set size is for this task?

Thanks again, and looking forward to your response!

**Comment by YifeiZhou02** [comment]
Hi Ciaran,

Thanks for your interest in our work. To confirm, we used `Llama-3.1-8B-Instruct.` `data_fraction` is set to 0.1 so that we only used 10% data for training the policy, this is just a hyperparameter where we found DPO can easily overfit using turn-wise advantage if too many gradient steps are performed. We suspect that there should be a better way to use more data but did not have a chance to look into it. For the backend task, the test set contains the first 1k examples. There are additional data in the Huggingface just in case they might be helpful for other purposes.

**Comment by CiaranZhou** [comment]
> Hi Ciaran,
> 
> Thanks for your interest in our work. To confirm, we used `Llama-3.1-8B-Instruct.` `data_fraction` is set to 0.1 so that we only used 10% data for training the policy, this is just a hyperparameter where we found DPO can easily overfit using turn-wise advantage if too many gradient steps are performed. We suspect that there should be a better way to use more data but did not have a chance to look into it. For the backend task, the test set contains the first 1k examples. There are additional data in the Huggingface just in case they might be helpful for other purposes.

Thanks for the helpful clarifications!

I had a quick follow-up question regarding the training of the critic model. In Table 5 of the paper, both the actor and the critic are trained with `nll_loss_coef` set to 0.01. However, based on the README, it seems that this parameter is not explicitly passed when training the critic, which (if using OpenRLHF's DPO implementation) would default it to 0.

Could you clarify whether `nll_loss_coef` was indeed set to 0.01 during critic training, or if the default value was used instead?

Thanks again for your time and help!

**Comment by YifeiZhou02** [comment]
I think the default value (of 0) should be able to reproduce the result in the paper as well, I did not find critic training to be very sensitive to `nll_loss_coef` but it might affect actor training sometimes.

**Comment by menyunlei** [comment]
hello， thanks for your great work!
I wonder if you can share the confige file of the model? Llama-3.1-8B-Instruct  and  Llama-3.1-70B-Instruct,  I download directly from Llama, missed those file.   

huggingface sometime do not work very well.

**Comment by YifeiZhou02** [comment]
Thanks for your interest. I was asked not to redistribute the model files.

Could you try checking other models on Huggingface such as Llama-3.1-8B (https://huggingface.co/meta-llama/Llama-3.1-8B/tree/main). This should have config.json in it if that is what you are looking for.
